Content.Client/UserInterface/Systems/Radial/Controls/RadialContainer.xaml.cs

```diff
@@ -0,0 +1,403 @@
+            ActionLabel.SetMarkup(CloseButton.Content ?? string.Empty);
+            ActionLabel.Visible = true;
+        };
+
+        CloseButton.Controller.OnMouseExited += (_) =>
+        {
+            PlaySizeAnimation(CloseButton, NormalSize, InSizeAnimationKey, OutSizeAnimationKey);
+            ActionLabel.Visible = false;
+        };
+        CloseButton.Controller.OnPressed += (_) =>
+        {
+            Dispose();
+        };
+    }
+
+    private void PlayRadialAnimation(Control button, Vector2 pos, string playKey)
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(MoveAnimationTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.Offset),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(new Vector2(0,0), 0f),
+                        new AnimationTrackProperty.KeyFrame(pos, MoveAnimationTime)
+                    }
+                }
+            }
+        };
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    private void PlaySizeAnimation(Control button, float size, string playKey, string? stopKey)
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(FocusAnimationTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.ButtonSize),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(button.Size, 0f),
+                        new AnimationTrackProperty.KeyFrame(new Vector2(size, size), FocusAnimationTime)
+                    }
+                }
+            }
+        };
+
+        if (stopKey != null && button.HasRunningAnimation(stopKey))
+            button.StopAnimation(stopKey);
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        base.Draw(handle);
+
+        foreach (var child in Layout.Children)
+        {
+            var button = (RadialItem)child;
+            LayoutContainer.SetPosition(child, button.Offset - (button.Size/2));
+        }
+
+        // FIXME: We use item's offset like "local item position" for animation. Need make some better way to do it;
+        LayoutContainer.SetPosition(CloseButton, CloseButton.Offset - (CloseButton.Size/2));
+        LayoutContainer.SetPosition(ActionLabel,
+            new Vector2(0 - (GetTextWidth(ActionLabel) / 4), 18.5f + GetDistance(4.5f) ));
+    }
```

Comment:
This should just be checking if the animation is finished up front still rather than recalculating everything on every draw call.